### PR TITLE
Detect libcpp

### PR DESCRIPTION
**Pull request by jschueller**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -132,6 +134,16 @@ if (WITH_THREADLOCAL AND NOT DEFINED HAVE_THREAD_LOCAL_STORAGE)
   endforeach()
 endif ()
 
+if (WITH_CXX OR BUILD_PYTHON OR BUILD_GUILE OR BUILD_OCTAVE)
+  check_cxx_symbol_exists (_LIBCPP_VERSION string SYSTEM_HAS_LIBCPP)
```
Comment (stevengj) [inline CMakeLists.txt]: Why do we need to check this?

Comment (stevengj) [inline CMakeLists.txt]: i.e. why not always use `-std=c++11` if we are compiling with C++ and that flag is supported?

Comment (jschueller) [inline CMakeLists.txt]: - c++ support is only enabled if requested or bindings enabled
- NLopt itself doesn't use c++11

Comment (jschueller) [inline CMakeLists.txt]: There's no problem with nlopt at all, I followed your remark about my travis script that has to set the -std=c++11 flag to compile on osx.

**Pull request merged**